packages/client/src/client/open-feature-client.ts

```diff
@@ -44,7 +48,9 @@ export class OpenFeatureClient implements Client {
 
   get metadata(): ClientMetadata {
     return {
-      name: this.options.name,
+      // Use domain if name is not provided
+      name: this.options.domain ?? this.options.name,
+      domain: this.options.domain ?? this.options.name,
```

Comment:
Maybe we should throw in the constructor if name and domain are supplied?
I am not sure but it could be confusing if someone outs in a name and a domain and the domain is set as name.
This would not break existing implementations but also keep people from using both.

Reply:
I could be missing something, but people shouldn't be constructing `OpenFeatureClient` anyway, and with your recent change they won't be able to, so I don't think this matters so much. This is basically internal, used by `OpenFeature.getClient()`, which just takes `domain` as a string now instead of name.

Reply:
I think you're probably right about that, ya.

Reply:
Yeah, I considered that but it could technically be viewed as a breaking change if someone was using the class. I'm happy to remove it if we think that's okay.

Reply:
What do you think @lukas-reining @toddbaert?

Reply:
If you want to merge this now, you should keep it fully non-beaking I think.

Reply:
Yes then lets prefer to release it and not remove the name from the options.
We can remove it later then when we do the breaking 2.0 release. Especially as the `OpenFeatureClient` class will then be removed from the exports anyways.
As we discussed already both changes should not be breaking conceptually but we can not guarantee technically.